library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -710,6 +711,23 @@ public void emitTuples()
   }
 
   /**
+   * Notifies that the directory is being scanned.<br>
+   * Override this method to custom handling. Will be called once
+   */
+  protected void visitDirectory(Path filePath)
+  {
+  }
+
+  private void checkVisitedDirectory(Path path)
+  {
+    String pathString = path.toString();
+    if (!processedFiles.contains(pathString)) {
```

Comment:
Since scanner is pluggable and user might implement there own scanner and may not necessary check for the path properly in there scanner. So this check provides additional safety.